iwyu_tool.py

```diff
@@ -308,6 +308,10 @@ def parse_compilation_db(compilation_db_path):
 
     # Expand symlinks.
     for entry in compilation_db:
+        # Convert relative paths to absolute ones if possible.
+        if 'directory' in entry.keys() and not os.path.isabs(entry['file']):
```

Comment:
Can remove `.keys()`, that's the default for `in` on `dict`s

Reply:
I've updated the commit message, removed .keys() and also added a test.

I went for an on-disk compliation database, hope that's OK. The alternative would be mocking __builtins__.open, but this is python2, so I can't use python3 unittest.mock, and doing that manually sounds boring. :-)